### PR TITLE
Adding the enum string for permissions and updating the mandatory parameters for Key Vault management

### PR DESCRIPTION
**Pull request by pomortaz**
Adding the enum string for permissions and updating the mandatory parameters for Key Vault management

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@pomortaz__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (pomortaz). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by kirthik** [comment]
test this please

File: src/ResourceManagement/KeyVaultManagement/Microsoft.Azure.Management.KeyVault/Generated/IKeyVaultManagementClient.cs [hunk 1]
```diff
@@ -1,7 +1,3 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (kirthik) [inline src/ResourceManagement/KeyVaultManagement/Microsoft.Azure.Management.KeyVault/Generated/IKeyVaultManagementClient.cs]: Why is this part removed in this file and the ones below?

Comment (pomortaz) [inline src/ResourceManagement/KeyVaultManagement/Microsoft.Azure.Management.KeyVault/Generated/IKeyVaultManagementClient.cs]: AutoRest tool removed the header comment, when I regenerated the code.

Comment (pomortaz) [inline src/ResourceManagement/KeyVaultManagement/Microsoft.Azure.Management.KeyVault/Generated/IKeyVaultManagementClient.cs]: @kirthik thanks. The header parameter was not set. Fixed it and updated the commit. :)

**Comment by kirthik** [comment]
test this please
